tfhe/src/integer/ciphertext/boolean_value.rs

```diff
@@ -0,0 +1,191 @@
+use super::{IntegerCiphertext, IntegerCiphertextRef, IntegerRadixCiphertext};
+use crate::integer::{RadixCiphertext, ServerKey};
+use crate::shortint::Ciphertext;
+use serde::{Deserialize, Serialize};
+
+/// Wrapper type used to signal that the inner value encrypts 0 or 1
+///
+/// Since values ares encrypted, it is not possible to know whether a
+/// ciphertext encrypts a boolean value (0 or 1). However some algorithms
+/// require that the ciphertext does indeed encrypt a boolean value.
+///
+/// This wrapper serves as making it explicit that it is known that the value
+/// encrypted is 0 or 1. And that if a function taking a BooleanValue as input
+/// returns incorrect value, it may be due to the value not really being 0 or 1.
+///
+/// Also some function such as comparisons are know to return an encrypted value
+/// that is either 0 or 1, and thus return a Ciphertext wrapped in a [BooleanValue].
+///
+/// [BooleanValue] can be used to wrap both owned types (`T`) and references (`&T`).
+///
+/// # Examples
+/// ## Converting a [BooleanValue] to a [RadixCiphertext]
+///
+/// ```
+/// use tfhe::integer::{gen_keys_radix, BooleanValue};
+/// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+///
+/// // We have 4 * 2 = 8 bits of message
+/// let size = 4;
+/// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size);
+///
+/// let a = 128u8;
+/// let b = 55u8;
+///
+/// let ct_a = cks.encrypt(a);
+/// let ct_b = cks.encrypt(b);
+///
+/// let ct_res = sks.ge_parallelized(&ct_a, &ct_b);
+/// // Convert the boolean value to a RadixCiphertext of size blocks
+/// // so we can use it in operations
+/// let ct_res = ct_res.into_radix(size, &sks);
+/// // Decrypt:
+/// let dec: u8 = cks.decrypt(&ct_res);
+/// assert_eq!(u8::from(a >= b), dec);
+/// ```
+///
+/// ## Wrapping a ciphertext in a BooleanValue
+/// ```
+/// use tfhe::integer::{gen_keys_radix, BooleanValue};
+/// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+///
+/// // We have 4 * 2 = 8 bits of message
+/// let size = 4;
+/// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size);
+///
+/// let clear_condition = 1u8;
+/// let a = 128u8;
+/// let b = 55u8;
+///
+/// let ct_a = cks.encrypt(a);
+/// let ct_b = cks.encrypt(b);
+/// let ct_condition = cks.encrypt(clear_condition);
```

Comment:
isn't it better to advertise encrypt_boolean here ?

Reply:
Yes it makes sense

Reply:
Actually no because the example is about showing we can wrap a regular radix ciphertext